src/eth1spec/evm.py

```diff
@@ -106,16 +115,33 @@ def process_call(
         depth=depth,
         env=env,
     )
-
-    # code = get_code(evm.env.state, evm.current)
-
-    return (evm.gas_left, [])
-
-    #  while(pc < len(code)):
-    #      op = code[pc]
-
-    #      switch(op):
-    #          case ADD:
-    #              x = evm.stack.pop()
-    #              y = evm.stack.pop()
-    #              evm.stack.append(x + y)
+    account = evm.env.state.get(evm.current, EMPTY_ACCOUNT)
+    code = account.code
+
+    if evm.value != 0:
+        evm.env.state[evm.caller].balance -= evm.value
+        evm.env.state[evm.current].balance += evm.value
+
+    while evm.pc < len(code):
+        op = bytes([code[evm.pc]])
+
+        if op == ops["ADD"]:
+            evm.gas_left -= gas_schedule["ADD"]
```

Comment:
Is it not the case that we should throw an error if the gas_left value becomes negative after this operation?

Reply:
there are many things still not checked around here :(